Added more validity for messages (#371)
FuelLabs/fuel-specs: specs/protocol/tx_validity.md

Write-destroy access list:

- For each [input `InputType.Coin`](./tx_format.md#inputcoin)
  - The [UTXO ID](./identifiers.md#utxo-id) `(txID, outputIndex)`
- For each [input `InputType.Contract`](./tx_format.md#inputcontract)
  - The [UTXO ID](./identifiers.md#utxo-id) `(txID, outputIndex)`
- For each [input `InputType.Message`](./tx_format.md#inputmessage)

>   - The [Input Message ID](./identifiers.md#input-message-id) `messageID`

Reviewer: ```suggestion
  - The [input message ID](./identifiers.md#input-message-id) `messageID`
```

Write-create access list:

- For each [output `OutputType.ContractCreated`](./tx_format.md#outputcontractcreated)
  - The contract ID `contractID`
- For each output
  - The [created UTXO ID](./identifiers.md#utxo-id)

Note that block proposers use the contract ID `contractID` for inputs and outputs of type [`InputType.Contract`](./tx_format.md#inputcontract) and [`OutputType.Contract`](./tx_format.md#outputcontract) rather than the pair of `txID` and `outputIndex`.

> Note that [output `OutputType.Message`](./tx_format.md#outputmessage) do not have a [UTXO ID](./identifiers.md#utxo-id) `(txID, outputIndex)` and are unspendable.

Reviewer: ```suggestion
Note that [`OutputType.Message` outputs](./tx_format.md#outputmessage) do not have a [UTXO ID](./identifiers.md#utxo-id), and are unspendable.
```

    if input.type == InputType.Contract:
        if not input.contractID in contracts:
                return False
    elif input.type == InputType.Message:

>         if not input.messageID in messages:

Reviewer: Above it says

> For a transaction tx, UTXO set state, and contract set contracts, the following checks must pass.

Can you add message set in there?

## Predicate Verification

> For each input of type `InputType.Coin` (or `InputType.Message`) and `predicateLength > 0`, [verify its predicate](../vm/main.md#predicate-verification).

Reviewer: ```suggestion
For each input of type `InputType.Coin` or `InputType.Message`, and `predicateLength > 0`, [verify its predicate](../vm/main.md#predicate-verification).
```
